src/modules/powerrename/lib/Settings.cpp

```diff
@@ -52,34 +57,208 @@ namespace
         SetRegNumber(valueName, value ? 1 : 0);
     }
 
-    std::wstring GetRegString(const std::wstring& valueName) {
+    std::wstring GetRegString(const std::wstring& valueName,const std::wstring& subPath)
+    {
         wchar_t value[CSettings::MAX_INPUT_STRING_LEN];
         value[0] = L'\0';
         DWORD type = REG_SZ;
         DWORD size = CSettings::MAX_INPUT_STRING_LEN * sizeof(wchar_t);
-        if (SUCCEEDED(HRESULT_FROM_WIN32(SHGetValue(HKEY_CURRENT_USER, c_rootRegPath, valueName.c_str(), &type, value, &size) == ERROR_SUCCESS)))
-        {
-            return std::wstring(value);
-        }
-        return std::wstring{};
+        std::wstring completePath = std::wstring(c_rootRegPath) + subPath;
+        SHGetValue(HKEY_CURRENT_USER, completePath.c_str(), valueName.c_str(), &type, value, &size);

```

Comment:
Let's keep the error checking.

Reply:
Added check if return value is `ERROR_SUCCESS`.